src/RA_md5factory.cpp

```diff
@@ -2,65 +2,56 @@
 
 #include "RA_Defs.h"
 
-namespace {
-const static unsigned int MD5_STRING_LEN = 32;
-}
+_CONSTANT_VAR MD5_STRING_LEN = 32;
 
 std::string RAGenerateMD5(const std::string& sStringToMD5)
 {
-    static char buffer[33];
-
     md5_state_t pms{};
-    md5_byte_t digest[16]{};
+    std::array<md5_byte_t, 16> digest{};
 
-    md5_byte_t* pDataBuffer = new md5_byte_t[sStringToMD5.length()];
+    auto pDataBuffer = std::make_unique<md5_byte_t[]>(sStringToMD5.length());
     ASSERT(pDataBuffer != nullptr);
     if (pDataBuffer == nullptr)
-        return "";
+        return {};
 
-    memcpy(pDataBuffer, sStringToMD5.c_str(), sStringToMD5.length());
+    std::memcpy(pDataBuffer.get(), sStringToMD5.c_str(), sStringToMD5.length());
 
     md5_init(&pms);
-    md5_append(&pms, pDataBuffer, sStringToMD5.length());
-    md5_finish(&pms, digest);
-
-    memset(buffer, 0, MD5_STRING_LEN + 1);
-    sprintf_s(buffer, MD5_STRING_LEN + 1,
-        "%02x%02x%02x%02x%02x%02x%02x%02x"
-        "%02x%02x%02x%02x%02x%02x%02x%02x",
-        digest[0], digest[1], digest[2], digest[3], digest[4], digest[5], digest[6], digest[7],
-        digest[8], digest[9], digest[10], digest[11], digest[12], digest[13], digest[14], digest[15]);
-
-    delete[] pDataBuffer;
-    pDataBuffer = nullptr;
-
-    return buffer;
+    md5_append(&pms, pDataBuffer.get(), sStringToMD5.length());
+    md5_finish(&pms, digest.data());
+
+    std::array<char, MD5_STRING_LEN + 1> buffer{};
+    Ensures(sprintf_s(buffer.data(), buffer.size(),
```

Comment:
Using `StringPrintf` with `std::string` seems to dangle the stored pointer, using `std::array` didn't dangle it though.